pulp-glue/pulp_glue/common/openapi.py

```diff
@@ -57,18 +84,36 @@ def basic_auth(self) -> t.Optional[t.Union[t.Tuple[str, str], requests.auth.Auth
         """Implement this to provide means of http basic auth."""
         return None
 
+    def auth(self, flow: t.Any) -> t.Optional[t.Union[t.Tuple[str, str], requests.auth.AuthBase]]:
+        """Implement this to provide other authentication methods."""
+        return None
+
```

Comment:
in search for a better name...
And the docstring should reflect what authentication this is for.

Reply:
Here's the thing: we will have to add a new method every time we want to add a new authentication method here? With this aren't we changing the interface of this class?

Reply:
That'd be an additive change.